### PR TITLE
Adds new more relaxed validator for Azure entities

### PR DESCRIPTION
**Pull request by danielestevez**

### CODE REVIEW

**Comment by nacx** [comment]
Please add unit tests that verify the validator and test the corner cases.

**Comment by danielestevez** [comment]
Added a test! (and fixed an error message by the way) with all the main cases i could think of @nacx 
Also, i added this class under _org.jclouds.azurecompute.arm.config.*_ not sure if it's the best place so feel free to suggest any other one

**Comment by nacx** [comment]
Pushed to master and 2.1.x. Thanks!

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Validates name for azure entities
+ * https://docs.microsoft.com/en-us/azure/architecture/best-practices/naming-conventions
+ *
+ * @see org.jclouds.predicates.Validator
+ */
+@Singleton
+public class AzureNameValidator extends Validator<String> {
+   private final int min = 2;
+   private final int max = 63;
```
Comment (demobox) [inline providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: [minor] Rename to make the purpose clearer, e.g. `minLength` and `maxLength`. Could these be static variables?

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+   public void validate(String name) {
+
+      if (name == null || name.length() < min || name.length() > max)
+         throw exception(name, "Can't be null or empty. Length must be " + min + " to " + max + " symbols.");
+      if (CharMatcher.JAVA_LETTER_OR_DIGIT.indexIn(name) != 0)
```
Comment (demobox) [inline providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: [minor] `!CharMatcher.JAVA_LETTER_OR_DIGIT.matches(name.charAt(0))` expresses the intent more clearly?

File: providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+         throw exception(name, "Should start with letter/number");
+
+      CharMatcher range = getAcceptableRange();
+      if (!range.matchesAllOf(name))
+         throw exception(name, "Should have lowercase or uppercase ASCII letters, numbers, or dashes");
```
Comment (demobox) [inline providers/azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureNameValidator.java]: What about `.` and `_`? If I'm understanding `getAcceptableRange` correctly, these are also allowed?

**Comment by danielestevez** [comment]
Thanks for the suggestions @demobox, PR was already merged but i added those improvements here https://github.com/jclouds/jclouds/pull/1215
